packages/integration-tests/src/test-project-setup/seed_test_project.ts

```diff
@@ -0,0 +1,226 @@
+      {
+        cwd: this.projectDirPath,
+        env: environment,
+      },
+    );
+
+    const startingInd = seedPolicyProcess.stdout.indexOf('{');
+    const cleanedPolicyString = seedPolicyProcess.stdout.slice(startingInd);
+
+    const clientConfig = await generateClientConfig(backendIdentifier, '1.3');
+    if (!clientConfig.custom) {
+      throw new Error('Client config missing custom section');
+    }
+    const seedRoleArn = clientConfig.custom.seedRoleArn as string;
+
+    const seedCredentials = await this.stsClient.send(
+      new AssumeRoleCommand({
+        RoleArn: seedRoleArn,
+        RoleSessionName: `seedSession`,
+        Policy: cleanedPolicyString,
+        PolicyArns: [
+          {
+            arn: 'arn:aws:iam::aws:policy/service-role/AmplifyBackendDeployFullAccess',
+          },
+        ],
+      }),
+    );
+
+    assert.ok(seedCredentials.Credentials);
+    assert.ok(seedCredentials.Credentials.AccessKeyId);
+    assert.ok(seedCredentials.Credentials.SessionToken);
+    assert.ok(seedCredentials.Credentials.SecretAccessKey);
+
+    await ampxCli(['sandbox', 'seed'], this.projectDirPath, {
+      env: {
+        AWS_ACCESS_KEY_ID: seedCredentials.Credentials!.AccessKeyId,
+        AWS_SECRET_ACCESS_KEY: seedCredentials.Credentials!.SecretAccessKey,
+        AWS_SESSION_TOKEN: seedCredentials.Credentials!.SessionToken,
+        ...environment,
+      },
+    }).run();
+  }
+
+  override async assertPostDeployment(
+    backendId: BackendIdentifier,
+  ): Promise<void> {
+    await super.assertPostDeployment(backendId);
+    const testUsername = 'testUser@testing.com';
+    const clientConfig = await generateClientConfig(backendId, '1.3');
+
+    if (!clientConfig.auth) {
+      throw new Error('Client config missing auth section');
+    }
+
+    if (!clientConfig.data) {
+      throw new Error('Client config missing data section');
+    }
+
+    const authenticatedUserCredentials =
+      await new AmplifyAuthCredentialsFactory(
+        this.cognitoIdentityProviderClient,
+        clientConfig.auth,
+      ).getNewAuthenticatedUserCredentials();
+
+    const httpLink = new HttpLink({ uri: clientConfig.data.url });
+    const link = ApolloLink.from([
+      createAuthLink({
+        url: clientConfig.data?.url,
+        region: clientConfig.data?.aws_region,
+        auth: {
+          type: AUTH_TYPE.AWS_IAM,
+          credentials: authenticatedUserCredentials.iamCredentials,
+        },
+      }),
+      httpLink,
+    ]);
+
+    const apolloClient = new ApolloClient({
+      link: link,
+      cache: new InMemoryCache(),
+    });
+
+    const content = await apolloClient.query({
+      query: gql`
+        query TestQuery {
+          listTodos {
+            items {
+              id
+              content
+            }
+          }
+        }
+      `,
+    });
+
+    assert.strictEqual(
+      content.data.listTodos.items[0].content,
+      `Todo list item for ${testUsername}`,
+    );
+  }
```

Comment:
The Todos created with cognito auth should have `owner` field populated.

Could you add `owner` filed to the list call and assert that it's non-empty string?